isotope/service/pkg/srv/request.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this currentFile except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package srv
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"istio.io/fortio/log"
+	"istio.io/tools/isotope/convert/pkg/consts"
+	"istio.io/tools/isotope/convert/pkg/graph/size"
+	"istio.io/tools/isotope/convert/pkg/graph/svctype"
+)
+
+func sendRequest(
+	destName string,
+	destType svctype.ServiceType,
+	size size.ByteSize,
+	requestHeader http.Header) (*http.Response, error) {
+	url := fmt.Sprintf("http://%s:%v", destName, consts.ServicePort)
+	request, err := buildRequest(url, size, requestHeader)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("sending request to %s (%s)", destName, url)
+	return http.DefaultClient.Do(request)
+}
+
+func buildRequest(
```

Comment:
likewise this could use fhttp/http_client.go

on the other hand I guess this is small/self contained

Reply:
If I recall correctly, I tried to switch to using the fhttp package but the extra features in it caused a slight slow down as far as max QPS. And if I don't need those features I'd rather not add the package.

Reply:
interesting in my tests, which I haven't rerun recently, the fast client was at least 30-50% faster and less resource intensive than the standard go client; but that's only when you reuse over multiple connections which is more tricky to do in a proxy mode